src/background/cookie-instrument.ts

```diff
@@ -1,87 +1,76 @@
-  // Instrument cookie changes
-  browser.cookies.onChanged.addListener(function(changeInfo) {
-    // Ignore requests made by extensions
-    /*
-      if (details.originUrl.indexOf("moz-extension://") > -1) {
-        return;
-      }
-      */
-    console.log(
-      "Cookie changed: " +
-        "\n * Cookie: " +
-        JSON.stringify(changeInfo.cookie) +
-        "\n * Cause: " +
-        changeInfo.cause +
-        "\n * Removed: " +
-        changeInfo.removed,
-      changeInfo,
-    );
-  });
+  constructor(dataReceiver) {
+    this.dataReceiver = dataReceiver;
+  }
+
+  public run(crawlID) {
+    // Instrument cookie changes
+    this.onChangedListener = async (changeInfo: {
+      /** True if a cookie was removed. */
+      removed: boolean;
+      /** Information about the cookie that was set or removed. */
+      cookie: Cookie;
+      /** The underlying reason behind the cookie's change. */
+      cause: OnChangedCause;
+    }) => {
+      const change = changeInfo.removed ? "deleted" : "added";
 
-  events.on(
-    "cookie-changed",
-    function(event) {
-      const data = event.data;
       // TODO: Support other cookie operations
-      if (data === "deleted" || data === "added" || data === "changed") {
-        const update = {};
-        update.change = loggingDB.escapeString(data);
-        update.crawl_id = crawlID;
+      if (change === "deleted" || change === "added" || change === "changed") {
+        const update = {} as JavascriptCookieChange;
 
-        let cookie = event.subject.QueryInterface(Ci.nsICookie2);
+        update.change = change;
+        // TODO: Add changeInfo.cause
+        update.crawl_id = crawlID;
 
-        // Creation time (in microseconds)
-        const creationTime = new Date(cookie.creationTime / 1000); // requires milliseconds
-        update.creationTime = creationTime.toLocaleFormat("%Y-%m-%d %H:%M:%S");
+        const cookie: Cookie = changeInfo.cookie;
 
         // Expiry time (in seconds)
+        // A comment from pre-webextension code, which may still be valid:
         // May return ~Max(int64). I believe this is a session
         // cookie which doesn't expire. Sessions cookies with
         // non-max expiry time expire after session or at expiry.
-        const expiryTime = cookie.expiry; // returns seconds
-        if (expiryTime === 9223372036854776000) {
-          const expiryTimeString = "9999-12-31 23:59:59";
+        const expiryTime = cookie.expirationDate; // returns seconds
+        let expiryTimeString;
+        if (!cookie.expirationDate || expiryTime === 9223372036854776000) {
+          expiryTimeString = "9999-12-31 23:59:59";
         } else {
           const expiryTimeDate = new Date(expiryTime * 1000); // requires milliseconds
-          const expiryTimeString = expiryTimeDate.toLocaleFormat(
-            "%Y-%m-%d %H:%M:%S",
+          expiryTimeString = moment(expiryTimeDate).format(
+            "YYYY-MM-DD HH:mm:ss",
```

Comment:
@englehardt All other timestamps are using `.toISOString()` except this one. Is there a specific reason for this?

Reply:
(We can drop the moment dependency if we use .toISOString() here as well)

Reply:
Totally fine to use `toISOString()`! This is just an inconsistency we never noticed.

Reply:
Ok, great. :) I'll change this and also change `expiryTimeString` to the ISO format.